src/hdmf/build/objectmapper.py

```diff
@@ -955,40 +951,33 @@ def __add_groups(self, builder, groups, container, build_manager, source, export
-                if attr_name is not None:
-                    attr_value = self.get_attr_value(spec, container, build_manager)
-                    if any(isinstance(attr_value, t) for t in (list, tuple, set, dict)):
-                        it = iter(attr_value)
-                        if isinstance(attr_value, dict):
-                            it = iter(attr_value.values())
-                        for item in it:
-                            if isinstance(item, Container):
-                                self.__add_containers(sub_builder, spec, item, build_manager, source, container, export)
+                attr_value = self.get_attr_value(spec, container, build_manager)
+                if isinstance(attr_value, (list, tuple, set, dict)):
+                    if isinstance(attr_value, dict):
+                        attr_value = attr_value.values()
+                    for item in attr_value:
+                        if isinstance(item, Container):
+                            self.__add_containers(sub_builder, spec, item, build_manager, source, container, export)
                 self.__add_groups(sub_builder, spec.groups, container, build_manager, source, export)
                 empty = sub_builder.is_empty()
                 if not empty or (empty and isinstance(spec.quantity, int)):
                     if sub_builder.name not in builder.groups:
                         builder.set_group(sub_builder)
-            else:
-                if spec.data_type_def is not None:
-                    self.logger.debug("    Adding group for spec name: %s, %s: %s, %s: %s"
-                                      % (repr(spec.name),
-                                         spec.def_key(), repr(spec.data_type_def),
-                                         spec.inc_key(), repr(spec.data_type_inc)))
-                    attr_name = self.get_attribute(spec)
-                    if attr_name is not None:
-                        attr_value = getattr(container, attr_name, None)
-                        if attr_value is not None:
-                            self.__add_containers(builder, spec, attr_value, build_manager, source, container, export)
-                else:  # data_type_def is None and data_type_inc is not None
-                    self.logger.debug("    Adding group for spec name: %s, %s: %s, %s: %s"
-                                      % (repr(spec.name),
-                                         spec.def_key(), repr(spec.data_type_def),
-                                         spec.inc_key(), repr(spec.data_type_inc)))
-                    attr_value = self.get_attr_value(spec, container, build_manager)
+            elif spec.data_type_def is not None:
+                self.logger.debug("    Adding group for spec name: %s, %s: %s"
+                                  % (repr(spec.name), spec.def_key(), repr(spec.data_type_def)))
+                attr_name = self.get_attribute(spec)
```

Comment:
Did you mean to replace this with `self.get_attr_value`? I might be misremembering our real-time discussion

Reply:
Yes, that is in a separate PR (which I will push in a little bit).